idp_task1.cpp

```diff
@@ -2,15 +2,43 @@
 #include "socket_exception.hpp"
 #include <iostream>
 #include <string.h>
+#include <vector>
+#include <thread>
 
 static const int BUFFER_MAXLEN = 100;
 static const int LISTENQ = 16;
 static const int SERVER_PORT = 8877;
 static const char* ECHO_STR_PREFIX = "echo: ";
 static const int ECHO_STR_LEN = strlen(ECHO_STR_PREFIX);
 
+void listeningThread(Socket_sptr client_socket)
+{
+    int n = 0;
+    int maxlen = 100;
+    char buffer[maxlen];
+
+    char echo[maxlen + ECHO_STR_LEN];
+    strcpy(echo, ECHO_STR_PREFIX);
+    memset(buffer, 0, maxlen);
+
+    while ((n = client_socket->Recv(buffer, maxlen, 0)) > 0)
+    {
+        printf("received: %s", buffer);
+
+        strcat(echo, buffer);
+
+        client_socket->Send(echo, strlen(echo), 0);
+        memset(buffer, 0, maxlen);
+        memset(echo + ECHO_STR_LEN, 0, maxlen);
+    }
+
+    client_socket->Close();
+}
+
 void createTestTCPServer()
 {
+    std::vector<std::thread> listeningThreads;
+
```

Comment:
There is some 'thash' code inside of `while` loop below. I can't comment on it 'cause it was not changed in this PR.
Please remove it to keep the code clean.